src/main/java/com/alseed/todolist/Main.java

```diff
@@ -102,4 +128,47 @@ private static String toggle(String input){
             return "Необработанное исключение";
         }
     }
+
+    private static String search(String[] input){
+        return taskList.
+                stream()
+                .filter((s) -> s.getDescription().contains(input[1]))
+                .map(Main::printOutputCreator)
+                .collect(Collectors.joining("\n"));
+    }
+
+    private static String delete(String[] input){
+        try {
+            int id = Integer.parseInt(input[1]);
+            taskList.removeIf((s) -> s.getId()==id);
+            return "";
+        }
+        catch (NumberFormatException e){
+            return "Неверно указан аргумент";
+        }
+    }
+
+    private static String edit(String[] input){
+        try {
+            int id = Integer.parseInt(input[1]);
+            taskList.stream()
+                    .filter(s -> s.getId() == id)
+                    .forEach(s -> s.setDescription(concatArray(input, 2)));
+            return "";
+        }
+        catch (NumberFormatException e){
+            return "Неверно указан аргумент";
+        }
+        catch (Exception e){
+            return "Необработанное исключение";
+        }
+    }
+
+    private static String concatArray(String[] input, int nStart) {
```

Comment:
Тут я бы тоже использовал StreamAPI
```suggestion
    private static String concatArray(String[] input, int nStart) {
         return IntStream.range(nStart, input.length).mapToObj(i -> input[i]).collect(Collectors.joining(" "));
    }
```